app/services/provider_interface/setup_provider_relationship_permissions.rb

```diff
@@ -0,0 +1,30 @@
+module ProviderInterface
+  class PermissionsSetupError < StandardError; end
+
+  class SetupProviderRelationshipPermissions
+    def self.call(permissions_data = {})
+      ProviderRelationshipPermissions.transaction do
+        permissions_data.each do |id, permissions|
+          record = ProviderRelationshipPermissions.find(id)
+          raise PermissionsSetupError, 'Permissions record has already been setup' if record.setup_at.present?
```

Comment:
Hopefully this will never happen! It is theoretically possible to get stuck here if another user simultaneously sets up a subset of the current permissions, bc then the already-setup permission is stuck in the session, the form can't be submitted, and the user can't move on. But we'd learn about it from Sentry. And the intention is to have only one manage_orgs user per org at first. **tl;dr** I think this is fine